test/sched.py

```diff
@@ -49,14 +50,25 @@
 
 os.environ['PBS_JOBID'] = "1"
 
-PBSNODEFILE = tempfile.NamedTemporaryFile(delete=False)
-PBSNODEFILE.write("localhost\nlocalhost\n")
-PBSNODEFILE.close()
-os.environ['PBS_NODEFILE'] = PBSNODEFILE.name
+def set_env():
+    pbsnodefile = tempfile.NamedTemporaryFile(delete=False)
+    pbsnodefile.write("localhost\nlocalhost\n")
+    pbsnodefile.close()
+    os.environ['PBS_NODEFILE'] = pbsnodefile.name
+    os.environ['PBS_NUM_PPN'] = '1'
+
 
 class TestSched(unittest.TestCase):
     """tests for vsc.mympirun.mpi.sched functions"""
 
+    def setUp(self):
+        self.orig_environ = os.environ
+        set_env()
+
+    def tearDown(self):
+        """Clean up after running test."""
+        os.environ = self.orig_environ
```

Comment:
also clean up the file in `$PBS_NODEFILE`, maybe have a `cleanup_PBS_env` too, which accepts `self.orig_environ` as parameter